SwanCustomEnvironments/swancustomenvironments/scripts/makenv.sh

```diff
@@ -171,16 +183,28 @@ _log "ENV_NAME:${ENV_NAME}"
 # Make sure the Jupyter server finds the new environment kernel in /home/$USER/.local
 mkdir -p /home/$USER/.local/share/jupyter/kernels
 ln -f -s ${ENV_PATH}/share/jupyter/kernels/${ENV_NAME} /home/$USER/.local/share/jupyter/kernels/${ENV_NAME} | tee -a "${LOG_FILE}"
+# Avoid an ipykernel installation warning message
+export JUPYTER_PATH=${ENV_PATH}/share/jupyter:$JUPYTER_PATH
 
 # Activate the environment
 _log "Setting up the environment..."
-source ${ENV_PATH}/bin/activate
+if [ -n "$CONDA_ENV" ]; then
```

Comment:
I think this would look nicer without so many if else.
What I proposed when we talked yesterday is to have individual scripts that the main makenv.sh script sources.

For example, so far I have identified three actions that differ depending on the environment type: create, activate, install. You could factor out the code that is specific for each builder into the auxiliary scripts.

This means you could have auxiliary scripts of the form:
- create_conda.sh, activate_venv.sh, activate_accpy.sh
- activate_conda.sh, ...

Or, alternatively, just one big script for each that does all the steps:
- conda.sh, venv.sh, accpy.sh

And the main script would just do:
`source ${builder_name}.sh` or `source create_${builder_name}.sh` etc.

And the main script would:

Reply:
Those scripts could be in a subdirectory called `builders`.

Reply:
Wouldn't we lose some of the flexibility that a single script provide and create some redundancy? Since some of the actions are done for several cases

Reply:
It depends on how much redundancy we are talking about. Modularity has a lot of value when structuring code: if you need to change something specific to e.g. coda, you know where to go. I don't think there's some much redundancy and, if necessary, we can factor out shared code in further auxiliary scripts.

In the end this is mimicking polymorphism. Would you like a code that does:
```
if object.class == Dog: object.run_function_dog()
else if object.class == Bear: object.run_function_bear()
...
```
or you like better
```
object.run_function() --> polimorphism will call one implementation or the other depending on runtime type of object
```
?